test/e2e/scheduling/predicates.go

```diff
@@ -825,6 +825,7 @@ func CreateHostPortPods(f *framework.Framework, id string, replicas int, expectR
 
 // create pod which using hostport on the specified node according to the nodeSelector
 func createHostPortPodOnNode(f *framework.Framework, podName, ns, hostIP string, port int32, protocol v1.Protocol, nodeSelector map[string]string, expectScheduled bool) {
+	hostIP = framework.TranslateIPv4ToIPv6(hostIP)
```

Comment:
This is a bit weird IMO, we're determining IP family by checking the cluster IP but we are validating host IP with it. Can we confirm that the host IP is always matching whatever the primary cluster IP is?

Reply:
the problem with these tests are they are using hardcoded IPv4 addresses.
The `TranslateIPv4ToIPv6(hostIP)` function returns the hostIP as IPv6, only if the cluster family is IPv6, otherwise returns hostIP without any change (as an IPv4 address)

Reply:
But this would break if an e2e test ends up passing in a v6 address for hostIP here right? We should account for that here or at the very least document that.

Reply:
ahh, ok got it, you are right, I was only focused on these tests since `createHostPortonNode` is not exported but I think I can modify the TranslateIPv4ToIPv6 to verify that the IP passed as argument is IPv4, otherwise do nothing

Reply:
done